src/SaaS.SDK.SalesforceIntegration/Functions.cs

```diff
@@ -0,0 +1,66 @@
+﻿using Microsoft.Azure.WebJobs;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.Marketplace.SaasKit.Client.DataAccess.Context;
+using Microsoft.Marketplace.SaasKit.Client.DataAccess.Contracts;
+using Microsoft.Marketplace.SaasKit.Client.DataAccess.Services;
+using Newtonsoft.Json;
+using SaaS.SDK.PublisherSolution.Models;
+using SaaS.SDK.PublisherSolution.Services;
+using System;
+using System.IO;
+using System.Net.Http;
+
+namespace SaaS.SDK.SalesforceIntegration
+{
+    public class Functions
+    {
+        public IConfiguration configuration { get; }
+
+        public Functions(IConfiguration configuration)
+        {
+            this.configuration = configuration;
+        }
+
+        public void ProcessQueueMessage([QueueTrigger("saas-process-salesforce")] string message, ILogger logger)
+        {
+            logger.LogInformation("In Process Queue Message with params : {0}",message);
+            try
+            {
+                var serviceProvider = IntitalizeRepositories();
+                QueueMessageParams queueMessageParams = JsonConvert.DeserializeObject<QueueMessageParams>(message);
+                QueueMessage messageParams = JsonConvert.DeserializeObject<QueueMessage>(queueMessageParams.Message);
+                HttpClient client = new HttpClient();
+                Salesforce sf = new Salesforce(serviceProvider,this.configuration);
+                logger.LogInformation("In Process -> Calling Authentication method to obtain access token");
+                var tokenResponse = sf.Authentication(client).ConfigureAwait(false).GetAwaiter().GetResult();
+                logger.LogInformation("In Process -> Access token obtain successfully");
+                client.DefaultRequestHeaders.Add("Authorization", "Bearer " + tokenResponse.AccessToken);
+                logger.LogInformation("In Process -> Calling QuoteAggregator");
+                var Response_ = sf.QuoteAggregator(client, messageParams,message).ConfigureAwait(false).GetAwaiter().GetResult();
+                logger.LogInformation("In Process -> Response received : {0}",JsonConvert.SerializeObject(messageParams));
+            }
+            catch (Exception ex)
+            {
+                logger.LogError($"Error occurred while processing request, Exception: {ex.InnerException}");
+                StorageQueueService sq = new StorageQueueService(this.configuration);
```

Comment:
Make sure that we are not enqueueing the message in the case of a failure that is not related to Salesforce.